tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_matrix_frob_norm_kk.hpp

```diff
@@ -0,0 +1,53 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_MATRIX_FROB_NORM_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_MATRIX_FROB_NORM_HPP_
+
+#include "signal_kokkos_impl_called.hpp"
+
+namespace KokkosKernelsSTD {
+
+template<
+    class ExeSpace,
+    class ElementType,
+    std::experimental::extents<>::size_type numRows,
+    std::experimental::extents<>::size_type numCols,
+    class Layout,
+    class Scalar>
+Scalar matrix_frob_norm(kokkos_exec<ExeSpace> kexe,
+			std::experimental::mdspan<
+			ElementType,
+			std::experimental::extents<numRows, numCols>,
+			Layout,
+			std::experimental::default_accessor<ElementType>> A,
+			Scalar init)
+{
+
+  Impl::signal_kokkos_impl_called("matrix_frob_norm");
+
+  // corner cases
+  constexpr std::size_t zero = 0;
+  constexpr std::size_t one  = 1;
+  if (A.extent(0) == zero || A.extent(1) == zero) {
+    return init;
+  }
+  else if(A.extent(0) == one && A.extent(1) == one) {
+    using std::abs;
+    return init + abs(A(0, 0));
+  }
+
+  const std::size_t totNumElements = A.extent(0)*A.extent(1);
+  using flatten_t = std::experimental::mdspan<
```

Comment:
[Required] This will only work correctly for `LayoutLeft` and `LayoutRight`, not for `LayoutStride`.  (In fact, historically, `LayoutLeft` and `LayoutRight` would introduce padding.  I think that's gone now, though.)  Would you consider at least constraining `Layout` to be either `LayoutLeft` or `LayoutRight`?  A fall-back implementation could just use a Kokkos mirror view.

Reply:
oh yes, you are right, thanks for catching this! i will fix it.

Reply:
ok, I think I fixed it. I am basically doing the scaled sum of square approach but without assuming any layout.